src/wallet/tx_builder.rs

```diff
@@ -347,6 +316,80 @@ impl<D: Database, Cs: CoinSelectionAlgorithm<D>> TxBuilder<D, Cs> {
     }
 }
 
+// methods supported only by create_tx
+impl<D: Database> TxBuilder<D, DefaultCoinSelectionAlgorithm, CreateTx> {
+    /// Create a builder starting from a list of recipients
+    pub fn with_recipients(recipients: Vec<(Script, u64)>) -> Self {
+        Self::default().set_recipients(recipients)
+    }
+
+    /// Replace the recipients already added with a new list
+    pub fn set_recipients(mut self, recipients: Vec<(Script, u64)>) -> Self {
+        self.recipients = recipients;
+        self
+    }
+
+    /// Add a recipient to the internal list
+    pub fn add_recipient(mut self, script_pubkey: Script, amount: u64) -> Self {
+        self.recipients.push((script_pubkey, amount));
+        self
+    }
+
+    /// Set a single recipient that will get all the selected funds minus the fee. No change will
+    /// be created
+    ///
+    /// This method overrides any recipient set with [`set_recipients`](Self::set_recipients) or
+    /// [`add_recipient`](Self::add_recipient).
+    ///
+    /// It can only be used in conjunction with [`drain_wallet`](Self::drain_wallet) to send the
+    /// entire content of the wallet (minus the `unspendable` UTXOs) to a single recipient or with a
+    /// list of manually selected UTXOs added with [`utxos`](Self::utxos) or
+    /// [`add_utxo`](Self::add_utxo).
+    ///
+    /// When bumping the fees of a transaction made with this option, the user should remeber to
+    /// add [`maintain_single_recipient`](Self::maintain_single_recipient) to correctly update the
+    /// single output instead of adding one more for the change.
+    pub fn set_single_recipient(mut self, recipient: Script) -> Self {
+        self.single_recipient = Some(recipient);
+        self.recipients.clear();
+
+        self
+    }
+
+    /// Enable signaling RBF
+    ///
+    /// This will use the default nSequence value of `0xFFFFFFFD`.
+    pub fn enable_rbf(self) -> Self {
+        self.enable_rbf_with_sequence(0xFFFFFFFD)
+    }
+
+    /// Enable signaling RBF with a specific nSequence value
+    ///
+    /// This can cause conflicts if the wallet's descriptors contain an "older" (OP_CSV) operator
+    /// and the given `nsequence` is lower than the CSV value.
+    ///
+    /// If the `nsequence` is higher than `0xFFFFFFFD` an error will be thrown, since it would not
+    /// be a valid nSequence to signal RBF.
+    pub fn enable_rbf_with_sequence(mut self, nsequence: u32) -> Self {
+        self.rbf = Some(nsequence);
+        self
+    }
+}
+
+// methods supported only by bump_fee
+impl<D: Database> TxBuilder<D, DefaultCoinSelectionAlgorithm, BumpFee> {
+    /// Bump the fees of a transaction made with [`set_single_recipient`](Self::set_single_recipient)
+    ///
+    /// In case extra inputs will be added while bumping the fees, instead of adding a change
+    /// output the only exsiting output will be updated.
```

Comment:
Consider re-writing this sentence. I think what it wants to say is:

```suggestion
    /// Even if extra inputs are added to bump the fee, a change output will not be added;
    /// the existing output will simply grow in value.
```

This is an interesting idea. Shouldn't you just always lower the existing output value to increase the fee of a `set_single_recipient`. That's probably what they wanted right?

Reply:
Well, not always: for instance, if you create a normal transaction (without `set_single_recipient`) and you manage to pick just the right inputs so that you don't have to create a change (basically what BnB tries to do), then your initial transaction would only have one output, but that output could be potentially for a payment of a specific value.

If you want to bump the fees of this transaction you should be careful not to touch that output, because then it could make your payment fail. Instead you would add more inputs and create a change, unless you can consume entirely the new inputs in fees.

Reply:
Basically, you want to keep track of whether an output was created as a payment or due to other intentions (consolidation, change, fanout, …). For example if the wallet gets used by a custodial exchange, maybe one user would withdraw (unknowingly) to another user of the same service. Even though this is may end up being a self-send by the wallet, the amount is crucial, since it will be credited as a deposit.

Reply:
So as discussed a bit on discord I think I am still unsure about this behavior. The correct behavior to me is that **you never add extra inputs to bump the fee** if `maintain_single_recipient` is set. If it's not possible to reduce the single output anymore to get enough fee then it should probably just return an error.

Reply:
I have changed the comment to this:

> Unless extra inputs are specified with `add_utxo` or `utxos`, this flag will make `bump_fee` reduce the value of the existing output, or fail if it would be consumed entirely given the higher new fee rate.
>
> If extra inputs are added and they are not entirely consumed in fees, a change output will not be added; the existing output will simply grow in value.